Add PG14 support and upgrade recommendations for Cloud

charislam opened the pull request:
# Description

- Timescale Cloud now supports PG14
- One-click major upgrading is also available for PG14
- Best practices for upgrades: forking and replicas

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)
- [ ] Version 1.7
- [ ] Older [specify]

# Links

Fixes https://github.com/timescale/docs-private/issues/24

charislam committed:
Add PG14 and upgrade recommendations 1 of 1

charislam committed:
Add PG14 and upgrade recommendations 1 of 1

Rajakavitha1 commented on cloud/maintenance.md:
```diff
@@ -107,7 +109,28 @@ creating a fork of your service, see the
+*   Try out the upgrade on the fork before running it on your production system.
+    This gives you a good idea of what happens during the upgrade, and how long it
+    might take.
+*   Keep a copy of your old database version if you're worried about losing it
```
Creating a fork and creating a copy of the old database are two different procedures. I would list them as three bullet points under best practices with the links to relevant pages to avoid redundancy:
* Create a fork 
* Back up the database
* Enable replication

Rajakavitha1 commented on cloud/maintenance.md:
```diff
@@ -107,7 +109,28 @@ creating a fork of your service, see the
 
 </procedure>
 
+### Best practices for upgrading
```
I would move Best practices before the upgrading

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

charislam committed:
Apply suggestions from review

charislam committed:
Apply suggestions from review

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

charislam committed:
Merge branch 'latest' into charis/cloud-pg14

charislam committed:
Comment out screenshot

charislam committed:
Comment out screenshot

charislam committed:
Merge branch 'latest' into charis/cloud-pg14

charislam committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

charislam committed:
Merge branch 'charis/cloud-pg14' of github.com:timescale/docs into charis/cloud-pg14

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

charislam committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity commented:
@charislam Can we merge this one now?

charislam commented:
> @charislam Can we merge this one now?

I don't think the feature is available yet, they're working on it but it's still on the launch calendar!

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

charislam commented on cloud/service-operations/maintenance.md:
```diff
@@ -67,7 +67,69 @@ downtime is usually between 30&nbsp;seconds and 5&nbsp;minutes. We endeavor to
 [cloud-login]: https://cloud.timescale.com
 [operations-forking]: cloud/:currentVersion:/operations/#fork-a-service
 [postgres-relnotes]: https://www.postgresql.org/docs/release/
+[replicas]: /high-availability/
```
```suggestion
[replicas]: /cloud/:currentVersion:/high-availability/
```

charislam committed:
Specify upgrade version and fix links

Loquacity commented on cloud/service-operations/maintenance.md:
```diff
@@ -67,7 +67,70 @@ downtime is usually between 30&nbsp;seconds and 5&nbsp;minutes. We endeavor to
+</highlight>
+
+### Recommended practices for upgrading
+Follow these optional recommendations for a smooth upgrade experience.
```
```suggestion
Follow these optional recommendations for a smooth upgrade experience:
```

Loquacity commented on cloud/service-operations/maintenance.md:
```diff
@@ -67,7 +67,70 @@ downtime is usually between 30&nbsp;seconds and 5&nbsp;minutes. We endeavor to
+*   Keep a copy of your database with your old version and data, if you're
+    worried about losing it. You can fork your database without upgrading the
+    fork to keep a duplicate Timescale Cloud service.
+*   Enable replicas for near zero-downtime-upgrades. If you have replicas
```
```suggestion
*   Enable replicas for near-zero-downtime upgrades. If you have replicas
```

Loquacity commented on cloud/service-operations/maintenance.md:
```diff
@@ -67,7 +67,70 @@ downtime is usually between 30&nbsp;seconds and 5&nbsp;minutes. We endeavor to
+    fork to keep a duplicate Timescale Cloud service.
+*   Enable replicas for near zero-downtime-upgrades. If you have replicas
+    enabled on Timescale Cloud, the upgrade process upgrades your nodes one by
+    one. Thus, one node is always available. The only downtime is the time
```
```suggestion
    one, so one node is always available. The only downtime is the time
```

Loquacity commented on cloud/service-operations/maintenance.md:
```diff
@@ -67,7 +67,70 @@ downtime is usually between 30&nbsp;seconds and 5&nbsp;minutes. We endeavor to
+*   Enable replicas for near zero-downtime-upgrades. If you have replicas
+    enabled on Timescale Cloud, the upgrade process upgrades your nodes one by
+    one. Thus, one node is always available. The only downtime is the time
+    needed for failover, which usually takes several seconds. For more
```
```suggestion
    needed for failover, which usually takes a few seconds. For more
```

Loquacity reviewed (commented):
Just small things

charislam committed:
Apply suggestions from code review

Co-authored-by: Lana Brindley <github@lanabrindley.com>

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity commented:
@ggodeke Can this merge now?

ggodeke commented:
Sorry @Loquacity, not quite yet! We are hoping to go live in the next two weeks. Thanks for your patience!

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

ggodeke committed:
Fixed some replicas info

Unfortunately replicas do not help reduce downtime during a major version upgrade at this time

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity committed:
Merge branch 'latest' into charis/cloud-pg14

Loquacity commented:
> Sorry @Loquacity, not quite yet! We are hoping to go live in the next two weeks. Thanks for your patience!

@ggodeke Ready to merge?

ggodeke commented:
We are almost ready to merge this!! @charislam One thing I am not sure how to do is it looks like this might overwrite the page and remove the [replicas and maintenance section](https://docs.timescale.com/cloud/latest/service-operations/maintenance/#replicas-and-maintenance). Is there something I need to do here to merge in that?

charislam committed:
Merge branch 'latest' into charis/cloud-pg14

charislam commented:
> We are almost ready to merge this!! @charislam One thing I am not sure how to do is it looks like this might overwrite the page and remove the [replicas and maintenance section](https://docs.timescale.com/cloud/latest/service-operations/maintenance/#replicas-and-maintenance). Is there something I need to do here to merge in that?

Just needs a merge conflict resolution! I think I handled it but double-check that everything is in the order you want :)

charislam committed:
Add missing links

ggodeke committed:
Merge branch 'latest' into charis/cloud-pg14
